pkg/evaluation/evaluator.go

```diff
@@ -362,3 +380,29 @@ func extractSkipAnnotations(configuration extractor.Configuration) map[string]st
 
 	return skipAnnotations
 }
+
+func updateFailedRuleLine(schemaPath string, yamlNode yaml.Node) (failedErrorLine int, failedErrorColumn int) {
```

Comment:
now that this method is **pure** we can (and should) add a unit test for it, there is some complex logic here